src/wire_msg.rs

```diff
@@ -41,7 +42,12 @@ impl WireMsg {
 
         let msg_header = MsgHeader::from_bytes(header_bytes);
         log::debug!("reading data of length: {}", msg_header.data_len());
-        let mut data: Vec<u8> = vec![0; msg_header.data_len()];
+        // https://github.com/rust-lang/rust/issues/70460 for work on a cleaner alternative:
+        #[cfg(not(any(target_pointer_width = "32", target_pointer_width = "64")))]
+        {
+            compile_error!("You need an architecture capable of addressing 32 bit pointers");
+        }
+        let mut data: Vec<u8> = vec![0; usize::try_from(msg_header.data_len()).unwrap()];
```

Comment:
We try to avoid using `.unwrap()` wherever we can to prevent panics.

Reply:
Thanks for taking a look!
This _should_ be an infallible operation on `32` and `64` bit platforms. The only alternative I can think of is to wait on https://github.com/rust-lang/rust/issues/70460, but if you know of another way, I'd love to hear it.

Reply:
Thanks for the PR, nice one. Where the type is a result then `?` is OK here. 
Where we see a result we always use it or pass an error back (or resolve the error).

Reply:
I added in a runtime check as well. Do we want to keep both the compile-time check and the (theoretically unreachable) runtime check?